sklearn/utils/estimator_checks.py

```diff
@@ -397,6 +398,39 @@ def check_dtype_object(name, Estimator):
 
 
 @ignore_warnings
+def check_dict(name, Estimator):
+    rnd = np.random.RandomState(0)
+    if name in ['SpectralCoclustering']:
+        return 0
+    if name in ['RANSACRegressor']:
+        X = 3 * rnd.uniform(size=(20, 3))
+    else:
+        X = 2 * rnd.uniform(size=(20, 3))
+
+    y = X[:, 0].astype(np.int)
+    y = multioutput_estimator_convert_y_2d(name, y)
+    estimator = Estimator()
+    set_testing_parameters(estimator)
+
+    if hasattr(estimator, "n_components"):
```

Comment:
Can't say anything here, I've seen the same part on several places, for example here: https://github.com/kiote/scikit-learn/blob/feature-test-__dict__/sklearn/utils/estimator_checks.py#L443

So, it looks like it could be moved there, do you think it should?